sample-compose/data/src/main/java/co/nimblehq/sample/compose/data/repository/AppPreferencesRepositoryImpl.kt

```diff
@@ -0,0 +1,44 @@
+package co.nimblehq.sample.compose.data.repository
+
+import android.util.Log
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.*
+import co.nimblehq.sample.compose.domain.repository.AppPreferencesRepository
+import kotlinx.coroutines.flow.*
+import java.io.IOException
+import javax.inject.Inject
+
+class AppPreferencesRepositoryImpl @Inject constructor(
+    private val appPreferencesDataStore: DataStore<Preferences>
+) : AppPreferencesRepository {
+
+    private object PreferencesKeys {
+        val FIRST_TIME_LAUNCH = booleanPreferencesKey("FIRST_TIME_LAUNCH")
+    }
+
+    override fun isFirstTimeLaunchPreferences(): Flow<Boolean> = flow {
+        val firstLaunchPreferences = appPreferencesDataStore.data
+            .catch { exception ->
+                if (exception is IOException) {
+                    Log.e(
+                        AppPreferencesRepositoryImpl::class.simpleName,
+                        "Error reading preferences.",
+                        exception
+                    )
```

Comment:
I think we can get rid of this log and replace it with a comment like what Google suggested https://developer.android.com/codelabs/android-preferences-datastore#5. Logging with `e` level will throw a critical message in red in the log system.

If we still want to log, let's use a lower level such as `d` or `i`. 🙏